### ISSUE TITLE
The mAP evaluation doesn't seem right

### ISSUE THREAD

**Issue by fernandocamargoti** [fizyr/keras-retinanet#715]
I've been using your library and noticed that the values of mAP were very different from the ones provided by the competition's score. At first, I thought it was just that I was overfitting train e val datasets. But I implemented a score_threshold search, in which I generate the mAP for each threshold. Then, I noticed that the smallest thresholds were always giving better mAP, what doesn't make sense. After having a look, I noticed that the implementation isn't taking false negatives into consideration. So, the recall formula was wrong, since it needs the false negatives.

I fixed the recall formula and it started penalizing for false negatives. But the scores were still very different from the ones in kaggle. So, digged deeper and noticed that it uses the curve of precision and recall to calculate the AP. I didn't understand why, but I decided to use the absolute values of true positives, false positives and false negatives in the formula provided in the competition: tp/(tp+fp+fn)

https://www.kaggle.com/c/rsna-pneumonia-detection-challenge#evaluation

Now, the scores make a lot more sense.

**Comment by EgorLakomkin** [comment]
Do you have a code or PR somewhere for the correct mAP calculations?

**Comment by fernandocamargoti** [comment]
I'll submit a PR with it.

**Comment by baljit92** [comment]
Hey @fernandocamargoti could you post the PR link over here, please?

**Comment by wfwiggins203** [comment]
@fernandocamargoti As I understand it, the RSNA Kaggle Challenge defines a different "precision" metric, as you stated above, then calculates the average "precision" over all IOU thresholds (which is **not** the same as "average precision"). In this sense, I don't think this is an issue with the keras-retinanet code base, but rather a quirk of the competition itself that necessitates modifications to achieve the correct metric in evaluating the training and performance of your model.

**Comment by hgaiser** [comment]
Closing this as it seems it's specifically for a challenge.
